application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java

```diff
@@ -487,6 +495,110 @@ public JdaTester() {
         return ErrorResponseException.create(reason, new Response(null, -1, "", -1, Set.of()));
     }
 
+    /**
+     * Creates a MessageReceivedEvent
+     * <p>
+     * The method creates a MessageReceivedEvent from the input message and mock the message author
+     * from the input user.
+     *
+     * @param message the message to create an event for
+     * @param user the author from the message
+     * @return the created MessageReceivedEvent
+     */
+    public MessageReceivedEvent createMessageReceivedEvent(@NotNull Message message,
+            @NotNull User user) {
+        doReturn(user).when(message).getAuthor();
+        return new MessageReceivedEvent(jda, MESSAGE_RESPONSE_NUMBER, message);
+    }
+
+    /**
+     * Creates a message spy
+     *
+     * @return the created spy
+     */
+    public @NotNull Message createMessageSpy() {
+        return spy(Message.class);
+    }
```

Comment:
Isnt there already a nice `MessageBuilder` in JDA? If so, we probably do not need utility for a dedicated spy, and the user can just use the builder.

Reply:
MessageBuilder doesnt work with new MessageReceivedEvent thats why i created a spy, but still i think it would be better if you could look into the JdaTester part, i try to close the other open conversations.